relayer/chain/ethereum/header_chain_state.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Snowfork
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package ethereum
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/tranvictor/ethashproof"
+	"golang.org/x/sync/errgroup"
+)
+
+type EthashproofCacheState struct {
+	sync.Mutex
+	currentCache *ethashproof.DatasetMerkleTreeCache
+	nextCache    *ethashproof.DatasetMerkleTreeCache
+}
+
+// HeaderChainState is a helper for tracking where we are
+// in the Ethereum chain. It warms up caches as we go.
```

Comment:
As mentioned in #166, how does this logic deal with different forks?

Reply:
I guess the ethash proof logic is fork independent, so that should be fine actually.

Reply:
Would it make sense to separate current/newest block (since that's fork dependent) from the ethashproofCache?

Reply:
I think it makes sense to separate tehm. My intuition was to keep this state together since current epoch is related to current forks. But it's not actually relevant to the implementation